main.cpp

```diff
@@ -309,13 +309,6 @@ int main(int argc, char* argv[]) {
         SDEBUG("Resetting database");
         string db = args["-db"];
         unlink(db.c_str());
-    } else if (args.isSet("-bootstrap")) {
-        // Allow for bootstraping a node with no database file in place.
-        SINFO("Loading in bootstrap mode, skipping check for database existance.");
-    } else {
-        // Otherwise verify the database exists
-        SDEBUG("Verifying database exists");
-        SASSERT(SFileExists(args["-db"]));
```

Comment:
Because in hc-tree mode, sqlite inspects the file to see if an existing file is hc-tree or not. An empty file then is not an hc-tree file, and we want to allow sqlite to create it.

There's really no reason to have this check in the first place, we get around it in various places by running `touch main.db`, which we might as well skip and just let bedrock make the file.

Reply:
Is you start without a valid database file and that flag is not present, I assume bedrock will crash... is the problem obvious in the logs without this check?

I assume the db flag check is to not meant for test cases where we touch main.db but for where someone starts bedrock by accident in a weird way, in which case, you could potentially add an `else if (args.isSet("-hctree")) { log something }`. Doing that also keeps the bootstrap log line, which is kinda handy.